### ISSUE TITLE
Feature: support accessibility escape gesture for going back or dismissing modal

### ISSUE THREAD

**Issue by m-abs** [NativeScript/NativeScript#4349]
To offer better support for accessibility for our iOS users, we need to be able to dismiss a modal or go back with the two-finger-Z gesture.

From https://developer.apple.com/library/content/featuredarticles/ViewControllerPGforiPhoneOS/SupportingAccessibility.html#//apple_ref/doc/uid/TP40007457-CH12-SW5

> Escape. A two-finger Z-shaped gesture that dismisses a modal dialog, or goes back one level in a navigation hierarchy.

To implement this, we need to override the native function "accessibiliityPerformEscape()".
This can be done on `UINavigationController` for going back one level in the history. I've still to figure out how to it for modals.

Adding these lines to the `UINavigationControllerImpl` in **ui/frame/frame.ios.ts** implements go-back:
```typescript
    public accessibiliityPerformEscape() {
      if (this.owner.canGoBack()) {
        this.owner.goBack();

        return true;
      }

      return false;
    }
```
It is only called if VioceOver is enabled.
If we can go back, goBack() is called and the function returns true, otherwise false.

Dismissing modals still needs to be worked out, but I'll be making a PR for this soon.

**Comment by m-abs** [comment]
I've started my work on this branch: https://github.com/m-abs/NativeScript/tree/feature/4349

**Comment by m-abs** [comment]
I can't run the nativescript tests on iOS, both master and my own branch crashes with this:

```bash
1   0x102d9527c -[TNSRuntime executeModule:referredBy:]
2   0x1028fbe00 main
3   0x106c4e92d start
4   0x1
[native code]: JS ERROR Error: Could not find module './'. Computed path '/Users/msn/Library/Developer/CoreSimulator/Devices/.../data/Containers/Bundle/Application/.../tests.app/app'.
Jun 14 15:40:28 wsnota313 com.apple.CoreSimulator.SimDevice.....launchd_sim[75276] (UIKitApplication:org.nativescript.tests[0x9553][76725]): Service exited due to signal: Segmentation fault: 11
```

**Comment by m-abs** [comment]
As explained in the PR, I'm closing this issue. https://github.com/NativeScript/NativeScript/pull/4382#issuecomment-312227388
